app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -315,4 +317,8 @@ public static String formLoadPayloadStatus() {
     public static boolean isDetailTabSwipeActionEnabled() {
         return doesPropertyMatch(DETAIL_TAB_SWIPE_ACTION_ENABLED, CommCarePreferences.YES, CommCarePreferences.YES);
     }
+
+    public static boolean isOfflineUpdateEnabled() {
+        return doesPropertyMatch(OFFLINE_UPDATE_ENABLED, CommCarePreferences.YES, CommCarePreferences.YES);
```

Comment:
Gotcha. @phillipm proposed making the offline install option only available when you are doing an update from within a user session, and never accessible from the app manager (with the exception of debug builds perhaps? just to make it easier for ourselves). Does that sound reasonable?